blueprint/core/blueprint_View.cpp

```diff
@@ -225,18 +252,93 @@ namespace blueprint
 
     void View::mouseDown (const juce::MouseEvent& e)
     {
-        dispatchViewEvent("onMouseDown", detail::makeViewEventObject(e, *this));
+        switch (e.source.getType())
+        {
+            case juce::MouseInputSource::InputSourceType::mouse:
+                dispatchViewEvent("onMouseDown", detail::makeViewEventObject(e, *this));
+                break;
+            case juce::MouseInputSource::InputSourceType::touch: {
+                Touch* t = new Touch(e.source.getIndex(), e.x, e.y, e.getScreenX(), e.getScreenY(), getViewId());
```

Comment:
So we're using raw `new` here and then storing the allocated pointer in a `juce::Array`. To my knowledge `juce::Array` is non-owning. As a result there is a memory leak here as I see no matching `delete` anywhere for these `Touch` objects. 

See `juce::Array` docs: 

"Holds a resizable array of primitive or copy-by-value objects"

I would question why we need to use `Touch*` here. `Touch` is a struct and not a polymorphic base or similar so this code should probably use a value type array `juce::Array<Touch>` and operate with references to a "copy-by-value" type.